httpx_retries/retry.py

```diff
@@ -65,12 +65,19 @@ class Retry:
             HTTPStatus.GATEWAY_TIMEOUT,
         ]
     )
+    RETRYABLE_EXCEPTIONS: Final[Tuple[Type[httpx.HTTPError], ...]] = (
+        httpx.ConnectTimeout,
+        httpx.NetworkError,
+        httpx.ProtocolError,
+        httpx.ReadTimeout,
+    )
```

Comment:
Having a look at https://www.python-httpx.org/exceptions/ , I'd suggest a different set of default exceptions:

- `httpx.TimeoutException`: all subclasses can be safely retried, so it's not necessary to specify `httpx.ConnectTimeout` and `httpx.ReadTimeout` individually.
- `httpx.RemoteProtocolError` should be used instead of `httpx.ProtocolError`, as `httpx.LocalProtocolError` can be caused by user error; in this case retries will always fail.

```suggestion
    RETRYABLE_EXCEPTIONS: Final[Tuple[Type[httpx.HTTPError], ...]] = (
        httpx.TimeoutException,
        httpx.NetworkError,
        httpx.RemoteProtocolError,
    )
```